### PR TITLE
merge selectivity to branch

### PR DESCRIPTION
**Pull request by birdstorm**
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tikv-client-lib-java/106)

### CODE REVIEW

File: src/main/java/com/pingcap/tikv/statistics/Histogram.java [hunk 1]
```diff
@@ -122,18 +121,18 @@ protected Histogram histogramFromStorage(
...
 
     this.id = colID;
     this.numberOfDistinctValue = distinct;
     this.lastUpdateVersion = lastUpdateVersion;
     this.nullCount = nullCount;
 
     int len = 0;
-    ArrayList<Bucket> tmpBuckets = new ArrayList<>(256);
-    for(int i = 0; i < 256; i ++) tmpBuckets.add(new Bucket());
+    ArrayList<Bucket> tmpBuckets = new ArrayList<>(4096);
```
Comment (zhexuany) [inline src/main/java/com/pingcap/tikv/statistics/Histogram.java]: Use List instead of ArrayList when declare a type.

Comment (zhexuany) [reply src/main/java/com/pingcap/tikv/statistics/Histogram.java]: Additionally, could you make this 4096 finalize?

File: src/main/java/com/pingcap/tikv/statistics/Histogram.java [hunk 2]
```diff
@@ -122,18 +121,18 @@ protected Histogram histogramFromStorage(
...
+    for(int i = 0; i < 4096; i ++) tmpBuckets.add(new Bucket());
```
Comment (zhexuany) [inline src/main/java/com/pingcap/tikv/statistics/Histogram.java]: When you decide make 4096 finalize, you can use that variable name here.

File: src/main/java/com/pingcap/tikv/statistics/Histogram.java [hunk 3]
```diff
@@ -154,10 +153,10 @@ protected Histogram histogramFromStorage(
         }
         bucket.setLowerBound(lowerBound);
         bucket.setLowerBound(upperBound);
+        System.out.println(lowerBound + " and " + upperBound);
       } catch (IndexOutOfBoundsException e) {
-        System.err.println("IndexOutOfBoundsException: " + e.getMessage());
-      } catch (Exception e) {
-        System.err.println("Exception: " + e.getMessage());
+        e.printStackTrace();
```
Comment (zhexuany) [inline src/main/java/com/pingcap/tikv/statistics/Histogram.java]: You can create Exception such as `TiHistgoramException`?

File: src/main/java/com/pingcap/tikv/statistics/Histogram.java [hunk 4]
```diff
@@ -279,7 +278,7 @@ protected int lowerBound(Comparable key) {
 
   // mergeBuckets is used to merge every two neighbor buckets.
```
Comment (zhexuany) [inline src/main/java/com/pingcap/tikv/statistics/Histogram.java]: Here, when you want to use comments, please follow google java style.

**Review by zhexuany — CHANGES_REQUESTED** [review]

**Comment by birdstorm** [comment]
Incorrect use of Comparables result in test failure.
- [x] rewriting Histogram logic to merge with #84

**Review by zhexuany — APPROVED** [review]

**Pull request merged**